infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLog.java

```diff
@@ -77,6 +79,15 @@
      */
     long getTrimMark();
 
+    /**
+     * Returns the missing addresses in this Log Unit in the specified consecutive
+     * range of addresses. The range cannot be larger than RECORDS_PER_LOG_FILE.
+     *
+     * @param range Range of addresses to compare.
+     * @return Set of missing addresses.
+     */
+    Set<Long> getMissingAddresses(Range<Long> range);
+
```

Comment:
I think this is more appropriate `List<Long> queryAddressesInRange(long lowerEndpoint, UpperEndpoint)`

Also, since the request size is not validated, I don't feel like it makes sense to mention RECORDS_PER_LOG_FILE. I think this limit should be set by the client anyways.

Reply:
Done.